lec12-numerical-models.Rmd

```diff
@@ -0,0 +1,902 @@
+- migration rate
+- abundance of food -->
+
+Let's make some sample data to help us come up with a simple model.
+
+For bacteria growing in a flask with plenty of food, we know that they divide in two periodically. 
+If we take one unit of time to be the time between divisions, we can assume that the population 
+approximately doubles at each time point.
+
+```{r}
+library(ggplot2) # ggplot2 includes the qplot function
+# qplot produces plots that are similar to ggplot, but you don't need a dataframe to use it.
+# We could use ggplot instead by first making a dataframe out of our data:
+# data <- data.frame(times, population_size)
+
+times <- seq(0, 10, by = 0.2) # make a sequence of time points
+population_size <- 2 ^ times # make a sequence for the population size: it doubles at every time point
+qplot(times, population_size) +
+  geom_line()
+```
+
+How do we write our population growth assumption as a general model? Let's calculate the difference in $N$ at each time point
+and plot this vs. $N$ at the previous time point. 
+
+```{r}
+# last N-1 points minus first N-1 points
+N_diff <- tail(population_size, -1) - head(population_size, -1) 
+
+qplot(head(population_size, -1), N_diff) +
+  geom_line()
+```
+
+This is a straight line, which tells us that the change in $N$ is proportional to $N$, 
+and in fact because of the way we defined time, the change in $N$ is exactly $N$. 
+
+Now we have our model: this model describes **exponential** growth of a population and is commonly used 
+for microbes growing with plenty of food. 
+
+$$\frac{dN}{dt}=N$$
+
+This is a **differential equation**: an equation that describes how some function of $N$ is related to
+derivatives of $N$. (In this case we'll just be using the first derivative.)
+
+If we wanted time to be something else, like minutes instead of generations, 
+we could add a constant of proportionality that captures how often our bacteria divide:
+
+$$\frac{dN}{dt} = rN$$
+
+This says that the population will increase by $r N$ per unit of time, 
+or the population increases by $N$ in time $1/r$. 
+
+The process we just used to get this model is probably not what you'd do in practice - 
+I just wanted to illustrate the connection between exponential growth and this differential equation.
+
+> #### Challenge
+>
+> If $r$ is made larger, does the population grow faster or slower? 
+
+## Analyzing models
+
+When you analyze a model, the exact questions you answer will depend on the model. 
+But in general, the following questions are ones you will try to answer for any population model.
+
+1. What are the **fixed points** of the model? (For which sizes of the population(s) does the population size remain constant in time?) 
+2. Which of the fixed points are **stable**? (If the population is slightly different from the fixed point, will it go towards it or away from it?)
+3. How do the above depend on parameters of the model?
+4. Which parameters fit my data best, and how good is the fit?
```

Comment:
Doesn't look like you go into detail here, but I wonder how this relates to the model fitting and selection we did in lecture 11. Might be a nice complement to think about (not in this lecture necessarily). Just curious.

Reply:
Yeah, I was trying to think how to compare these kinds of fits with the stuff you covered. I'm not going into any detail on Maximum Likelihood, but in theory you can just use ML to do the fitting I'm talking about and then calculate AIC as usual. I'll try to work up an example just for completeness, but probably won't cover it in class.